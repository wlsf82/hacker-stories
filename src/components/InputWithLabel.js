import React from 'react';

export const InputWithLabel = ({
  id,
  value,
  type = 'text',
  onInputChange,
  isFocused,
  children,
}) => {
  const inputRef = React.useRef();

  React.useEffect(() => {
    if (isFocused && inputRef.current) {
      inputRef.current.focus();
    }
  }, [isFocused])

  return (
    <>
      <label htmlFor={id}>{children}</label>
      <input
        ref={inputRef}
        id={id}
        type={type}
        value={value}
        onChange={onInputChange}
      />
    </>
  )
}
