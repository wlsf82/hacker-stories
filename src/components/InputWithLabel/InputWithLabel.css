.label {
  border-top: 1px solid #171212;
  border-left: 1px solid #171212;

  padding-left: 5px;
  font-size:24px;
}

.input {
  border: none;
  border-bottom: 1px solid #171212;
  background-color: transparent;

  font-size: 24px;
}
