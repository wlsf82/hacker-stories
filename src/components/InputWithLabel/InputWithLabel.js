import React from 'react';

import './InputWithLabel.css';

export const InputWithLabel = ({
  id,
  value,
  type = 'text',
  onInputChange,
  isFocused,
  children,
}) => {
  const inputRef = React.useRef();

  React.useEffect(() => {
    if (isFocused && inputRef.current) {
      inputRef.current.focus();
    }
  }, [isFocused]);

  return (
    <>
      <label htmlFor={id} className="label">{children}</label>
      &nbsp;
      <input
        ref={inputRef}
        id={id}
        type={type}
        value={value}
        onChange={onInputChange}
        className="input"
      />
    </>
  );
};
