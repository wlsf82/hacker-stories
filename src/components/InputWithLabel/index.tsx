import React from 'react';

import { InputWithLabelProps } from '../../types';

import './InputWithLabel.css';

export const InputWithLabel = ({
  id,
  value,
  type = 'text',
  onInputChange,
  isFocused,
  children,
}: InputWithLabelProps) => {
  const inputRef = React.useRef<HTMLInputElement>(null!);

  React.useEffect(() => {
    if (isFocused && inputRef.current) {
      inputRef.current.focus();
    }
  }, [isFocused]);

  return (
    <>
      <label htmlFor={id} className="label">{children}</label>
      &nbsp;
      <input
        ref={inputRef}
        id={id}
        type={type}
        value={value}
        onChange={onInputChange}
        className="input"
      />
    </>
  );
};
