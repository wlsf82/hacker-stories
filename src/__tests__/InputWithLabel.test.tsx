import React from 'react';
import renderer from 'react-test-renderer';

import InputWithLabel from '../components/InputWithLabel';

describe('InputWithLabel', () => {
  it('renders snapshot', () => {
    const inputWithLabelProps = {
      id: 'foo',
      value: 'Cypress',
      onInputChange: jest.fn(),
    };

    const component = renderer.create(
      <InputWithLabel {...inputWithLabelProps}>
        Cypress
      </InputWithLabel>
    )
    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
