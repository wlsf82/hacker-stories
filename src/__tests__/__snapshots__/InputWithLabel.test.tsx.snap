// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputWithLabel renders snapshot 1`] = `
Array [
  <label
    className="label"
    htmlFor="foo"
  >
    Cypress
  </label>,
  "Â ",
  <input
    className="input"
    id="foo"
    onChange={[MockFunction]}
    type="text"
    value="Cypress"
  />,
]
`;
